pkg/gofr/service/custom_header_test.go

```diff
@@ -0,0 +1,173 @@
+package service
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+
+	"gofr.dev/pkg/gofr/logging"
+)
+
+func Test_CustomDomainProvider_Get(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	path := "/path"
+	queryParams := map[string]interface{}{"key": "value"}
+	body := []byte("body")
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodGet, r.Method)
+
+		w.WriteHeader(http.StatusOK)
+
+		_, err := w.Write(body)
+		if err != nil {
+			return
+		}
+	}))
+	defer server.Close()
+
+	httpService := NewHTTPService(server.URL, logging.NewMockLogger(logging.INFO), nil,
```

Comment:
this is a shadow declaration of a `httpService` struct. We can rename this.
Please update in other places as well.